### PR TITLE
Fixes related to symlinks and legal files

### PR DESCRIPTION
**Pull request by michaeltrobinson**
Fixed issue where Godep would think files weren't tracked by VCS if the GOPATH had a symlink.
Fixed issue where LICENSE files within dependencies vendor directories would be re-vendored, resulting in failing builds.

### CODE REVIEW

File: save.go [hunk 1]
```diff
@@ -316,7 +325,7 @@ func copySrc(dir string, deps []Dependency) error {
 		w = fs.Walk(rootdir)
 		for w.Step() {
 			fname := filepath.Base(w.Path())
-			if IsLegalFile(fname) {
+			if IsLegalFile(fname) && !strings.Contains(w.Path(), sep) {
```
Comment (freeformz) [inline save.go]: Can you add a test case for this?

**Comment by freeformz** [comment]
It's probably best if you want to discuss these / work these through to split this PR into two. One for each issue.

PS: Symlinks inside $GOPATH aren't really supported (ex. https://groups.google.com/d/msg/golang-dev/GLUZGtUSj8A/SGUtCooLIT0J), so I'm curious as to how you are using them and why we would need to resolve them.

**Comment by michaeltrobinson** [comment]
Okay, I will add a test case and split this into two PRs with more details

**Comment by freeformz** [comment]
Thank you!

**Pull request closed without merging**